### PR TITLE
Pass setter schema for required setters check

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent @seans3 

This PR passes the setters schema to downstream kyaml library so that the schema is explicitly passed.

### CODE REVIEW

File: cmd/apply/cmdapply.go [hunk 1]
```diff
@@ -84,7 +83,7 @@ type ApplyRunner struct {
 }
 
 func (r *ApplyRunner) RunE(cmd *cobra.Command, args []string) error {
-	if err := setters2.CheckRequiredSettersSet(openapi.Schema()); err != nil {
+	if err := util.CheckForRequiredSetters(args[0]); err != nil {
```
Comment (monopole) [inline cmd/apply/cmdapply.go:86]: if len(args) < 1 ...?

File: cmd/preview/cmdpreview.go [hunk 2]
```diff
@@ -81,8 +80,7 @@ type PreviewRunner struct {
 
 // RunE is the function run from the cobra command.
 func (r *PreviewRunner) RunE(cmd *cobra.Command, args []string) error {
-	err := setters2.CheckRequiredSettersSet(openapi.Schema())
-	if err != nil {
+	if err := util.CheckForRequiredSetters(args[0]); err != nil {
```
Comment (monopole) [inline cmd/preview/cmdpreview.go:83]: ditto

File: cmd/util/util.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+	"sigs.k8s.io/kustomize/kyaml/setters2"
+)
+
+// KRMFileName returns the name of the KRM file. KRM file determines package
+// boundaries and contains the openapi information for a package.
+var KRMFileName = func() string {
```
Comment (mortent) [inline cmd/util/util.go:16]: Can't we use the value from `cmd/config'?

File: cmd/util/util.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	return "Krmfile"
+}
+
+// CheckForRequiredSetters takes the package path, checks if there is a KrmFile
+// and checks if all the required setters are set
+func CheckForRequiredSetters(path string) error {
+	krmFileName := KRMFileName()
+	krmFilePath := filepath.Join(path, krmFileName)
+	_, err := os.Stat(krmFilePath)
+	if !os.IsNotExist(err) {
```
Comment (mortent) [inline cmd/util/util.go:26]: What happens for other errors than `IsNotExist`?

Comment (phanimarupaka) [reply cmd/util/util.go]: Updated the logic.

File: cmd/util/util.go (hunk 3 above)
Comment (phanimarupaka) [reply cmd/util/util.go:16]: Shouldn't we add cmd/config as a dependency for cli-utils if we want to use that value ?

File: cmd/apply/cmdapply.go (hunk 1 above)
Comment (phanimarupaka) [reply cmd/apply/cmdapply.go]: Thanks. Added the check.

File: cmd/util/util.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 The Kubernetes Authors.
```
Comment (seans3) [inline cmd/util/util.go:1]: I'm not sure we should place a util library under the `cmd` hierarchy. Shouldn't `util.go` be under `pkg`

**Review by seans3 — CHANGES_REQUESTED** [review]
I'm not sure this file is in the correct place.

**Pull request closed without merging**